### PR TITLE
Update the Continous Integration documentation

### PR DESCRIPTION
**Pull request by robinsoc**
Add the Tuleap API Jenkins plugin documentation for
  - Pull request notification
  - Automated tests results

### CODE REVIEW

File: languages/en/user-guide/ci.rst [hunk 1]
```diff
@@ -62,6 +68,112 @@ an error message that will help to diagnose the problem.
...
+        stages {
+            stage('Build') {
+                steps {
+                    sh('make all')
+                }
+            }
+        }
+        post {
+            failure {
+                tuleapNotifyCommitStatus status: 'failure', repositoryId: '1', credentialId: '1234'
```
Comment (vaceletm) [inline languages/en/user-guide/ci.rst:118]: I don't think it's clear enough for the reader that 1234 refers to a jenkins credential.
You should add a step between token generation and this pipeline to create the credential and refer to it here

File: languages/en/user-guide/ci.rst [hunk 2]
```diff
@@ -62,6 +68,112 @@ an error message that will help to diagnose the problem.
...
+Here is an example Jenkinsfile using the tuleapSendTTMResults command:
+
+.. sourcecode:: groovy
+
+    pipeline {
+      agent any
+      stages {
+         stage('Reporting') {
+            steps {
+                tuleapSendTTMResults filesPath: 'outputs/*.xml', campaignId: '18', credentialId: 'tuleap-access-token'
```
Comment (vaceletm) [inline languages/en/user-guide/ci.rst:169]: It makes little sense to hardcode the campaign ID, it should be a job parameter and your pipleline should expose how to do it

File: languages/en/user-guide/ci.rst (hunk 1 above)
Comment (robinsoc) [reply languages/en/user-guide/ci.rst]: done

File: languages/en/user-guide/ci.rst (hunk 2 above)
Comment (robinsoc) [reply languages/en/user-guide/ci.rst]: done

**Pull request merged**